### ISSUE TITLE
Things in IPLD on IPFS

### ISSUE THREAD

**Issue by vorburger** [enola-dev/enola#777]
It may be fun to explore storing Enola's Things on [IPFS's IPLD](https://ipld.io).

This would first require being able to turn an `IImmutableThing` into the IPLD Data Model. That would enable to create predictable hashes of Things, to be generate CIDs. This will require work to turn any Datatype into one or several predictable 0/1 bits format/s (simplest could be e.g. just `toString`, and Maps as JSON with sorted keys?).

That would enable e.g. [IPLD DAG JSON Codec](https://ipld.io/specs/codecs/dag-json/spec/) as an [Enola Format](https://docs.enola.dev/concepts/core/#formats).

[IPFS IPLD Schemas](https://ipld.io/docs/schemas/intro/) would need to be an [Enola Schema](https://docs.enola.dev/concepts/core/#schemas).

Related libraries of possible interest:

* https://github.com/ipld/java-cid
* https://github.com/ipld/java-ipld-cbor 
* https://github.com/filip26/copper-multicodec 
* https://github.com/filip26/copper-multibase

**Comment by vorburger** [comment]
https://www.w3.org/2024/12/rch-wg-charter.html

**Comment by vorburger** [comment]
> https://openengiadina.codeberg.page/rdf-cbor/ but that's not what one would want... It could be a basis, but we would want (...)

https://codeberg.org/openEngiadina/rdf-cbor/issues/3

**Comment by vorburger** [comment]
https://lod-cloud.net/#ipfs (broken image) and https://lod-cloud.net/ipfs-how-to.html is not what this issue is primarily about (it's just RDF in [HDT](https://www.rdfhdt.org) on IPFS), but still worth documenting.

**Comment by bumblefudge** [comment]
would [this multicodec](https://github.com/multiformats/multicodec/blob/master/table.csv#L543) help?

**Comment by vorburger** [comment]
> would [this multicodec](https://github.com/multiformats/multicodec/blob/master/table.csv#L543) help?

Yes, that (Multicodec #543 `rdfc-1`, with code `0xb403`) [from what I understand...] does seem pertinent to this idea!

I hadn't noticed that before - Thank You very much for chiming in here and pointing that out.

I would probably use that - if I ever 🕰️ get around to actually implementing this...

**Comment by bumblefudge** [comment]
well, if I miss the github notification when you do, reach out at cal.com/ {username} or {username} at learningproof.xyz and we can chat!
